WorkspaceServer/WorkspaceFeatures/PackageTool.cs

```diff
@@ -31,21 +31,29 @@ public PackageTool(string name, IDirectoryAccessor directoryAccessor)
         public async Task<ProjectAsset> LocateProjectAsset()
         {
             var result = await CommandLine.Execute(_path.Value, MLS.PackageTool.PackageToolConstants.LocateProjectAsset, DirectoryAccessor.GetFullyQualifiedRoot());
-            var projectDirectory = new DirectoryInfo(string.Join("", result.Output));
-            return new ProjectAsset(new FileSystemDirectoryAccessor(projectDirectory));
+            if (result.ExitCode == 0)
+            {
+                var projectDirectory = new DirectoryInfo(string.Join("", result.Output));
+                return new ProjectAsset(new FileSystemDirectoryAccessor(projectDirectory));
+            }
+
+            return null;
         }
 
         public async Task<WebAssemblyAsset> LocateWasmAsset()
         {
             var result = await CommandLine.Execute(_path.Value, MLS.PackageTool.PackageToolConstants.LocateWasmAsset, DirectoryAccessor.GetFullyQualifiedRoot());
-            var projectDirectory = new DirectoryInfo(string.Join("", result.Output));
-
-            if (!projectDirectory.Exists)
+            if(result.ExitCode == 0)
             {
-                return null;
+                var projectDirectory = new DirectoryInfo(string.Join("", result.Output));
+
+                if (projectDirectory.Exists)
+                {
+                    return new WebAssemblyAsset(new FileSystemDirectoryAccessor(projectDirectory));
+                }
             }
 
-            return new WebAssemblyAsset(new FileSystemDirectoryAccessor(projectDirectory));
```

Comment:
Does the calling side handle returning null? Would it be better to throw here so that we can throw an informative exception (instead of NRE in the caller)?

Reply:
The caller here is not sure that the asset certainly exists and calls this function and if the function returns null it handles that https://github.com/dotnet/try/blob/3fedb8fa682242fa466ed2b826fb8df934a4e465/WorkspaceServer/Packaging/WebAssemblyAssetFinder.cs#L41

If we throw an exception here, should the caller catch the exception ?

Reply:
Whose error is it if the asset doesn't exist?

Reply:
Made this a little clearer with https://github.com/dotnet/try/pull/273